pkg/controller/controller.go

```diff
@@ -343,12 +343,18 @@ func (ctrl *resizeController) syncPVC(key string) error {
 		klog.V(4).Infof("No need to resize PVC %q", util.PVCKey(pvc))
 		return nil
 	}
-	if !ctrl.pvNeedResize(pvc, pv) {
-		klog.V(4).Infof("No need to resize PV %q", pv.Name)
-		return nil
-	}
 
-	return ctrl.resizePVC(pvc, pv)
+	if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+		_, _, err, _ := ctrl.expandAndRecover(pvc, pv)
```

Comment:
why return pv, pvc, but without using use objects?

Reply:
for now, mainly for tests.